tea.py

```diff
@@ -0,0 +1,443 @@
+#
+# TEA class
+# 
+
+import pybamm
+import matplotlib.pyplot as plt
+#import numpy as np
+import matplotlib.patches as patches
+import pandas as pd
+
+
+class TEA:
+    """
+    A Techno-Economic Analysis class for estimation of cell metrics:
+
+    Parameters
+    ----------
+    parameter_values : dict
+        A dictionary of parameters and their corresponding numerical values.
+        Default is NoneParameters
+    """
+
+    def __init__(self, parameter_values):
+        self.parameter_values = parameter_values
+
+    def print_stack_breakdown(self, print_dict=False):
+        if not hasattr(self, "stack_bd"):
+                self.calculate_stack_breakdown()
+        if print_dict == True:
+            print(dict)
+        # convert the stackbreakdown dict to a dataframe with three columns,
+        # in the first column the keys, in the second column the values for volume loadings
+        # and in the third column the values for mass loadings
+
+        # Create a dataframe with a column for the components, a column for the volume loadings, and a column for the mass loadings
```

Comment:
this whole block is quite repetitive and hard to read. i wonder if it's better to define a function that fills out an entry in volumes, masses and densities, and then loop over the values in components to do the calculations. you can also handle special cases here like 0 volume

Reply:
For example 
```python3
def get_density(component):
    mass_loading = stack_bd[f"{component} mass loading [mg.cm-2]"]
    volume_loading = stack_bd[f"{component} volume loading [uL.cm-2]"]
    if volume_loading == 0:
        density = 0
    else:
        density = mass_loading/volume_loading
    return density 
```
The dry electrode is a special case, but I think they should be handled by first defining the mass and volume for the dry electrode then using the same density calculation.

I actually think all calculations should happen in `calculate_stack_breakdown()`. Then you should just have a method to build and return the dataframe